bigtop-packages/src/common/logstash/do-component-build

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+. `dirname $0`/bigtop.bom
+
+export RELEASE=1
+export LANG="en_US.UTF-8"
+
+if [ -f /etc/os-release ]; then
+    . /etc/os-release
+fi
+
+case ${ID} in
+    ubuntu | debian)
+        sudo apt-get -y update
+        sudo apt-get install -y gnupg2
+        ;;
+esac
+
+case ${ID}-${VERSION_ID} in
+    ubuntu-18* | debian-9)
+    gpg2 --list-keys
+    echo "disable-ipv6" >> ~/.gnupg/dirmngr.conf
+    ;;
+esac
+
+gpg2 --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+curl -sSL https://get.rvm.io | bash -s stable
+[ -s /usr/local/rvm/scripts/rvm ] && source /usr/local/rvm/scripts/rvm
+
+case ${ID} in
+    ubuntu | debian)
+        rvm install ruby-2.4.0 && rvm use 2.4.0
+        ;;
+    fedora | centos)
+        rvm install ruby-2.3.8 && rvm use 2.3.8
```

Comment:
ditto. The installation.

Reply:
Yes, I agree that all the env specific packages should be installed by bigtop_toolchain. 
But `rvm` and `gpg2` are just used for logstash-package build, and the bigtop_toolchain is the common facility for all components of Bigtop. 
So how about to just install them during `do-components-build` of logstash?

Reply:
I think that not our convention. bigtop_toolchain is exactly how weinstall all the required dependencies, across all the components for bigtop bistro. Install things in do-components-build is rather not aligned. toolchain is also served the purpose of formalize the build env and make it static so that we can iterate the building code on top of it w/o being effected by env changes. 

Any other reason you'd like to put in in do-components-build?

Reply:
OK I see the comprehensive comparison in below. Thanks!